File: figures/09-string-matching/snippet-naive-match.java
    public static List<Integer> naiveAll(String text, String pattern) {
        int n = text.length(), m = pattern.length();
        if (m == 0) return List.of(0);
        List<Integer> out = new ArrayList<>();
        for (int s = 0; s + m <= n; s++) {
            int j = 0;
            while (j < m && text.charAt(s + j) == pattern.charAt(j)) j++;
            if (j == m) out.add(s);
        }
        return out;
    }

File: figures/09-string-matching/snippet-naive-match.py
def naive_match(text, pattern):
    n, m = len(text), len(pattern)
    if m == 0:
        return [0]
    result = []
    for s in range(n - m + 1):
        j = 0
        while j < m and text[s + j] == pattern[j]:
            j += 1
        if j == m:
            result.append(s)
    return result
